### DISCUSSION TITLE
DutyCycle of Raspberrymatic ... the sky is the limit

### DISCUSSION THREAD

**Discussion by The-Real-Lemonbiter** [SukramJ/aiohomematic#1786] [Q&A] [answered]
ENGLISH VERSION below: 

Ich bin neu bei Home Assistant (komme von ioBroker) und liebe die allgemeine Einfachheit von HA. Allerdings stehe ich vor einem riesigen Problem mit der Homematic-Integration über die custom_homematic Homematic(IP)-Integration für Home Assistant.

Etwa 15-20 Minuten nach Aktivierung der Integration schnellt der Duty Cycle meiner Raspberrymatic von 23 auf 95 oder manchmal sogar noch höher. Sobald ich die Integration deaktiviere, sinkt er wieder ab.

Zusätzliche Info: ioBroker läuft parallel auf derselben Raspberrymatic... Sind das vielleicht zu viele Anwendungen, die gleichzeitig an den Daten zupfen? Ich glaube nicht, weil ich den umgekehrten Test gemacht habe – das Deaktivieren der ioBroker Adapter senkt den Duty Cycle nicht...

Also, biiiiitte helft einem verzweifelten Neuling! 😅

Vielen Dank! LEM

I am new to HA (coming from ioBroker) and love the - in general - easyness of HA. On the other hand, I am having a massive issue with the integration of homematic using the custom_homematic Homematic(IP) Integration for Home Assistant.
After only about 15-20 minutes after activating the integration the duty cycle of my Raspberrymatic goes from 23- 95 or sometimes even higher. The other way around once I deactivate the integration.
As additional info: ioB is running and connected to the same raspberrymatic in parallel... is that too many applications grabbing the data? I do not believe so as I did the reverse testing, deactivating the instances in ioBroker does not reduce the duty Cycle... So pleazzzze help a newbie... 

Thanks
LEM

**Comment by SukramJ** [comment]
Deine Probleme mit dem Duty Cycle kann ich nicht nachvollziehen. Ich habe 3-4 HA-Installationen (180 Geräte) an meinem RM angeschlossen, ohne dass es zu besonderen Problemen gekommen wäre. Nach dem Start holt sich die Integration die benötigten Gerätewerte von der CCU, sofern diese nicht bereits aus dem CCU-Cache abgerufen werden konnten. Danach werden nur noch von der CCU gesendete Events mit Datenänderungen verarbeitet. Lediglich Systemvariablen und Programme werden zyklisch abgefragt, diese haben aber keinen Einfluss auf den Duty Cycle. Informationen zu Geräte-Firmware-Updates werden alle 6 Stunden abgefragt.

Woher Deine Duty Cycle-Probleme kommen, kann ich aber nicht erklären.

**Comment by The-Real-Lemonbiter** [comment]
Lieber Sukram, vielen Dank, so wie es bei Dir läuft, würde ich es auch erwarten... Danke für Deine Bestätigung also... Dennoch ist es bei mir so...
Hast Du viele batteriebetriebenen Geräte wie Fenstersensoren etc? Wenn ja, hast Du bei denen die Batterieladung aboniert oder wie das heisst?
 kann ich beim Einrichten der HACS Installation etwas verkehrt gemacht haben?

So sieht der Dutycyle bei mir aus: um 3:30 die Instanz gestartet, um 8:30 heute morgen wieder aus:
![image](https://github.com/user-attachments/assets/d46f94d3-f6c6-403f-9b93-e8ec76415bef)

**Reply by SukramJ** [reply]
> Hast Du viele batteriebetriebenen Geräte wie Fenstersensoren etc? 

Ja, ca. 70 % der Geräte sind batteriebetrieben.

> Wenn ja, hast Du bei denen die Batterieladung aboniert oder wie das heisst?

> > Danach werden nur noch von der CCU gesendete Events mit Datenänderungen verarbeitet

Wie geschrieben wird hier nicht aboniert.

> kann ich beim Einrichten der HACS Installation etwas verkehrt gemacht haben?

Nein.

**Reply by The-Real-Lemonbiter** [reply]
OK... wenn ich nichts verkehrt gemacht haben kann, dann muss ich mich also verabschieden vom Traum auf HA umzuziehen... ;-(

**Reply by SukramJ** [reply]
Da ich nicht von einem Problem mit der Integration ausgehe, vermute ich das Problem in deiner CCU bzw. deiner HM-Gerätelandschaft.
Ich würde mich an deiner Stell an das [Homematic-Forum](https://homematic-forum.de/forum/viewforum.php?f=84) wenden. Dort sind Experten, die sicherlich hilfreicher bei dem Thema sind als ich. Ggf. mit einer Rückverlinkung auf diese Diskussion.
Ein RM spezifische Problem würde ich hier nicht vermuten.

**Reply by The-Real-Lemonbiter** [reply]
ich glaube ich habe einen ersten Hiweis, was das verursacht: Sobald ich die versteckte Entität zur Batterieüberwachung aktiviere bei ein paar der Geräte, geht der DC nach oben: 
![image](https://github.com/user-attachments/assets/9f696784-b2c8-4340-a916-e953564415a2)

Ich habe das bei 6 Geräten mal manuell eingeschlaltet und der DC ist bereits vor mehr als 30 Minuten von 21 auf 56 gestiegen...und seitdem steht er da

![image](https://github.com/user-attachments/assets/7d5fc86e-9d85-472e-ac52-2427e405a436)

Wenn ich das nun bei allen 65 Batteriegeräten mache, dann ist halt "schluss mit lustig"

Und das passiert dann jedes Mal, wenn man HA neustarten muss, was gerade ja am Anfang wenn man vieles einbaut  immer mal wieder notwendig ist

**Reply by SukramJ** [reply]
Ich wiederhole noch mal:

> Nach dem Start holt sich die Integration die benötigten Gerätewerte von der CCU, sofern diese nicht bereits aus dem CCU-Cache abgerufen werden konnten. Danach werden nur noch von der CCU gesendete Events mit Datenänderungen verarbeitet. 

Selbst wenn der einmalige Abruf den Dutycycle für 65 Geräte durch Datenabruf erhöht, so wäre das nur eine einmalige Aktion. Wenn die CCU schon länger läuft, dann werden nur noch Daten aus dem Cache geladen und es findet kein Gerätezugriff mehr statt.  Danach wird da nichts mehr abgerufen. Bei dir bleibt der Dutycycle dauerhaft hoch.

**Reply by The-Real-Lemonbiter** [reply]
Vielen Dank  :-) Was ich nur noch einmal klären wollte war, ob diese versteckte Entität bei Dir eingeschaltet ist bei allen Geräten mit Batterie... Wenn nicht, dann könnte das ja der Grund sein, warum es bei Dir nicht auftritt.

**Reply by The-Real-Lemonbiter** [reply]
p.s.: Ich habe danke Deines Hinweises im HM Forum auch ein Ticket eröffnet

**Reply by The-Real-Lemonbiter** [reply]
Lieber Sukram, das lässt mir keine Ruhe... :-)
Ich habe es im Laufe des Tages noch am System bei mir in der Firma gestestet. Da ist das Raspberrymatic und das HA System läuft auf einem NAS als natives HA OS. Hier habe ich exakt das selbe Problem. Also andere CCU, anders leeres HA.
Ich denke, das Problem ist definitiv von der Abfrage der Batteriestati.
Aus meiner Sicht ist es wohl so: Jedes mal wenn ein Dashboard geöffnet wird, auf dem Batteriestati angezeigt werden, fragt HA bei der CCU nach und anstelle den in der CCU bekannten zurückzuliefern fragt sie das Gerät: "Na wie ist den Batteriestatus?" Sind dann auf einer Seite viele dieser Batterieinfos fragt sie bei jedem Gerät nach und KABUMM. 
In der Firma habe ich zwar nur 22 Geräte mit Batterie, aber selbst hier geht der DC auf 99 und dann stürzt HA ab, weil die Datenbank einen Fehler wirft:
<img width="554" alt="image" src="https://github.com/user-attachments/assets/3430168b-bd41-4f48-b387-47e85da16374">

Deshalb erlaube mir noch mal eine konkrete Rückfrage: Hast Du die versteckte Entität "Batteri" bei allen Batterien sichtbar geschaltet und mehrere davon auf einem Dashboard angezeigt? Bei mir habe ich zum Test ein reines Dashboard nur mit Batteriestati. Schon 10 reichen und der DC geht durch die Decke...
LG
LEM

**Reply by SukramJ** [reply]
> Ich denke, das Problem ist definitiv von der Abfrage der Batteriestati.

Ich habe keine Lust mich ständig zu wiederholen.

Bei mir werden 3 HA Instanzen gegen eine CCU betrieben, und bei allen sind nahezu alle Entitäten aktiviert.
Mein DC liegt nachts bei unter 10%.

**Reply by SukramJ** [reply]
Aktiviere auf der Integration doch den `debug` Modus und schau selber welche Kommunikation läuft.
Suche nach `getValue` und du siehst was noch individuell abgerufen wird.
Das `debug` Log siehst Du dann im HA Log (full log)

**Reply by The-Real-Lemonbiter** [reply]
Lieber Sukram, es tut mir leid, wenn Dich meine Rückfragen beleidigt haben! Ich wollte Dich nicht verärgern, aber Du hattest und hast meine Fragen leider nicht wirklich beantwortet und hinzu kommt, dass ich leider scheinbar nicht geschickt genug mich klar auszudrücken.

... "Jedes mal wenn ein Dashboard geöffnet wird, auf dem Batteriestati angezeigt werden, fragt HA bei der CCU nach und anstelle den in der CCU bekannten zurückzuliefern fragt sie das Gerät: "Na wie ist den Batteriestatus?" Sind dann auf einer Seite viele dieser Batterieinfos fragt sie bei jedem Gerät nach und KABUMM." ...

Damit meinte ich, dass ggf. möglicherweise die CCU den Fehler macht und diese Anfrage jedes mal an die Geräte sendet... Es kann aber auch an meiner CCU liegen, denn ich habe noch einen anderen bedauerlichen Fehler, den ich gerade zu analysieren versuche... 
Sobald ich etwas herausgefunden habe gebe ich sehr gerne Feedback

**Reply by The-Real-Lemonbiter** [reply]
@SukramJ Lieber Sukram: Ich habe den Fehler gefunden. Meine Raspberrymatic hat wenige Tage vorher durch ein defekte Wandthermometer Probleme Verursacht. Dieser Fehler hat sich fortgesetzt. Lange Rede kurzer Sinn, der Fehler lag in meiner Installation! Ich danke Dir sehr für Deine angebotene Hilfe!!!

Ich habe allerdings aktuell immer diese Fehlermeldung: Ping/Pong Übereinstimmungsfehler
Es besteht eine Diskrepanz zwischen gesendeten Ping-Ereignissen und empfangenen Pong-Ereignissen für die HA-Instanz HomeAss.

Das erklärt sich leier auch in der Doku nicht lösbar für mich. Aktuelle greifen zwar nicht zwei HA Installationen auf die CCU zu, wohl aber ioBroker und HA. Darf man ioB und HA nicht gleichzeitig auf die CCU3 (Raspberrymatic) zugreifen lassen oder muss ich da etwas beondres beachten?

Viele Grüße
LEM

**Reply by SukramJ** [reply]
Sehr schön.
Deine CCU kann dein HA auf dem NAS nicht erreichen. 
Ist da ggf. noch eine Firewall auf dem NAS aktiv, die eingehende Verbindungen blockiert?

**Reply by The-Real-Lemonbiter** [reply]
Danke für die schon wieder superschnelle ANtwort. Die Verbindung läuft immer für ein paar Stunden, ich kann aus HA die Komponenten sehen und steuern, dann iwann bricht die Verbindung ab und wenn ich dann in HA die Installation anhalte, dann hängt sich HA auf und braucht länger (so 3-4 Minuten), bis HA wieder erreichbar ist. Gleichzeitig steigt der DC auf der CCU. 
ioBroker ist per SSH an die CCU gebunden während HA über die normalen Ports angebunden ist.

Ein generelles nicht erreichen mag ich als ausschließen, denn die Verbindung funktioniert am Anfang ja. Werde aber nachher die Firewall auf meiner Unifi anschauen... 

Braucht es generell unterschiedliche Benutzernamen auf der CCU, einen für ioB und einen für HA? Oder müssen beide über dieselben Ports zugreifen?

**Reply by The-Real-Lemonbiter** [reply]
p.s.: Meine CCU ist ein Raspberrymatic auf eigener Hardware, HA läuft als HA OS in Proxmox

**Reply by SukramJ** [reply]
Ich nutze IOBroker nicht, also kann ich dazu nichts sagen.

> wenn ich dann in HA die Installation anhalte, dann hängt sich HA auf und braucht länger (so 3-4 Minuten), bis HA wieder erreichbar ist. Gleichzeitig steigt der DC auf der CCU.

Irgendwas ist immer noch sehr schräg bei dir.

> Braucht es generell unterschiedliche Benutzernamen auf der CCU

Nein.

Ohne Log Datei ist das alles nur rumgerate. Also bitte die Logdatei (full log) hier anhängen, sobald der Fehler wieder aufgetreten ist.
Und erstmal ohne `debug` laufen lassen.

**Reply by The-Real-Lemonbiter** [reply]
Danke, dass Du Dir das anschaust:

```
`2024-10-29 16:49:29.448 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration webrtc which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-10-29 16:49:29.448 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-10-29 16:49:29.448 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration homematicip_local which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-10-29 16:49:29.449 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration bambu_lab which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-10-29 16:49:29.752 WARNING (Recorder) [homeassistant.components.recorder.util] The system could not validate that the sqlite3 database at //config/home-assistant_v2.db was shutdown cleanly
2024-10-29 16:49:30.060 WARNING (Recorder) [homeassistant.components.recorder.util] Ended unfinished session (id=93 from 2024-10-29 12:48:05.216086)
2024-10-29 17:00:06.873 WARNING (MainThread) [homeassistant.components.sonos] Subscription to 192.168.2.207 failed, attempting to poll directly
2024-10-29 18:02:10.582 WARNING (MainThread) [hahomematic.platforms.hub.sensor] Value of sysvar SYS_SafeStatus exceedes maximum allowed length of 255 chars by Home Assistant. Value will be limited to 255 chars
2024-10-29 18:02:10.582 WARNING (MainThread) [hahomematic.platforms.hub.sensor] Value of sysvar SYS_SafeStatus exceedes maximum allowed length of 255 chars by Home Assistant. Value will be limited to 255 chars
2024-10-29 18:02:10.586 ERROR (MainThread) [homeassistant.components.sensor] Error adding entity sensor.ccu2_sv_dd_verbrauch_gestern for domain sensor with platform homematicip_local
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 662, in state
numerical_value = float(value) # type:ignore[arg-type]
^^^^^^^^^^^^
ValueError: could not convert string to float: '0.00 kWh, 0.00 Euro'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 595, in _async_add_entities
await coro
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 909, in _async_add_entity
await entity.add_to_platform_finish()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1366, in add_to_platform_finish
self.async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1005, in async_write_ha_state
self._async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1130, in _async_write_ha_state
self.__async_calculate_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1067, in __async_calculate_state
state = self._stringify_state(available)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1011, in _stringify_state
if (state := self.state) is None:
^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 664, in state
raise ValueError(
ValueError: Sensor sensor.ccu2_sv_dd_verbrauch_gestern has device class 'None', state class 'None' unit 'kWh' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: '0.00 kWh, 0.00 Euro' (<class 'str'>)
2024-10-29 18:02:10.597 ERROR (MainThread) [homeassistant.components.sensor] Error adding entity sensor.ccu2_sv_aa_verbrauch_gestern for domain sensor with platform homematicip_local
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 662, in state
numerical_value = float(value) # type:ignore[arg-type]
^^^^^^^^^^^^
ValueError: could not convert string to float: '0.08 kWh, 0.02 Euro'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 595, in _async_add_entities
await coro
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 909, in _async_add_entity
await entity.add_to_platform_finish()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1366, in add_to_platform_finish
self.async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1005, in async_write_ha_state
self._async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1130, in _async_write_ha_state
self.__async_calculate_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1067, in __async_calculate_state
state = self._stringify_state(available)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1011, in _stringify_state
if (state := self.state) is None:
^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 664, in state
raise ValueError(
ValueError: Sensor sensor.ccu2_sv_aa_verbrauch_gestern has device class 'None', state class 'None' unit 'kWh' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: '0.08 kWh, 0.02 Euro' (<class 'str'>)
2024-10-29 18:02:10.602 WARNING (MainThread) [hahomematic.platforms.hub.sensor] Value of sysvar SYS_SafeStatus exceedes maximum allowed length of 255 chars by Home Assistant. Value will be limited to 255 chars
2024-10-29 18:02:40.610 WARNING (MainThread) [hahomematic.platforms.entity] FIRE_entity_updated_EVENT failed: Sensor sensor.ccu2_sv_aa_verbrauch_gestern has device class 'None', state class 'None' unit 'kWh' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: '0.08 kWh, 0.02 Euro' (<class 'str'>)
2024-10-29 18:02:40.610 WARNING (MainThread) [hahomematic.platforms.entity] FIRE_entity_updated_EVENT failed: Sensor sensor.ccu2_sv_dd_verbrauch_gestern has device class 'None', state class 'None' unit 'kWh' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: '0.00 kWh, 0.00 Euro' (<class 'str'>)
2024-10-29 18:02:40.611 WARNING (MainThread) [hahomematic.platforms.hub.sensor] Value of sysvar SYS_SafeStatus exceedes maximum allowed length of 255 chars by Home Assistant. Value will be limited to 255 chars
2024-10-29 18:03:10.608 WARNING (MainThread) [hahomematic.platforms.entity] FIRE_entity_updated_EVENT failed: Sensor sensor.ccu2_sv_aa_verbrauch_gestern has device class 'None', state class 'None' unit 'kWh' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: '0.08 kWh, 0.02 Euro' (<class 'str'>)
2024-10-29 18:03:10.609 WARNING (MainThread) [hahomematic.platforms.entity] FIRE_entity_updated_EVENT failed: Sensor sensor.ccu2_sv_dd_verbrauch_gestern has device class 'None', state class 'None' unit 'kWh' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: '0.00 kWh, 0.00 Euro' (<class 'str'>)
2024-10-29 18:03:10.610 WARNING (MainThread) [hahomematic.platforms.hub.sensor] Value of sysvar SYS_SafeStatus exceedes maximum allowed length of 255 chars by Home Assistant. Value will be limited to 255 chars
2024-10-29 19:40:12.122 WARNING (MainThread) [homeassistant.components.sonos] Subscription to 192.168.2.210 failed, attempting to poll directly
2024-10-29 19:44:35.226 ERROR (MainThread) [homeassistant.components.shelly] Error fetching Shelly i3 Peter Büro data: Error fetching data: DeviceConnectionTimeoutError(TimeoutError())
`
```

**Reply by SukramJ** [reply]
Dein Log, das Du beim nächsten mal bitte als Datei anhängst, enthält keine Fehler die deine Probleme erklären.

**Reply by The-Real-Lemonbiter** [reply]
Ich muss nochmal dumm nachfragen. Meine CCU3 Raspberrymatic ist eigentlich abgeschottet mit diesen Einstellungen:
<img width="502" alt="image" src="https://github.com/user-attachments/assets/fd849321-3ddd-44a8-bcbb-ca1394839891">

Homeassitsant greift aktuell über die normalen Ports zu. So sind die Einstellungen:
<img width="407" alt="image" src="https://github.com/user-attachments/assets/528a7392-4c79-4318-ba61-3f16ee42bb71">

Wie greift die eigentlich zu?? Kann hier der Fehler liegen?

**Reply by SukramJ** [reply]
Du hast eine absolut 08/15 Standard Umgebung für HA und RM, die keiner besonderen Einstellung bedarf! 
Die zu öffnenden Firewallports sind in der Doku hinterlegt, und sehen bei dir richtig aus wobei ich Port 80 und 443 nicht sehe.
Über Sinn und Zweck dieser "Abschottung" wundere ich mich regelmäßig. Wenn Du da ein Problem siehst frag ich mich warum Du diese Firewall nicht einfach deaktiviert hast.
 
Es gibt keine **magische** Einstellung in dieser Integration, die eine CCU zum Ausflippen bringt.

**Reply by The-Real-Lemonbiter** [reply]
Ja ich verstehe das auch alles nicht. Habe Deinen Vorschlag gestern schon umgesetz und die Firewall in der CCU komplett ausgeschaltet quasi... danach dann die CCU neu gestartet... sah alles gut aus... Heute Nachts wieder das: Was Du siehst ist der DC meiner CCU

![image](https://github.com/user-attachments/assets/1229c2b8-3b7a-40e9-9547-4b4285a43269)

Als ich dann heute um 9:50 die Installation in HA abgeschaltet habe, ging der DC wieder ganz flott runter:

Wir kommen also so nicht weiter. Meine konkrete Frage: Wenn ich die Installation aus HACS lösche, wird dann alles im Zusammenhang mit der HomematicIP installation tatsächlich gelöscht? Oder muss / sollte ich etwas manuell machen? Ich würde gerne diese Installation von HomematicIP noch einmal von Scratch starten. 
Oder bleibt bei De-Installation immer noch wo etwas liegen?

LG
DANKE

**Reply by SukramJ** [reply]
Dach der De-Installation aus HACS solltest Du schauen ob unter `/config/custom_components` der order `custom_homematic` noch da ist und ggf. löschen.

> Ich habe den Fehler gefunden. Meine Raspberrymatic hat wenige Tage vorher durch ein defekte Wandthermometer Probleme Verursacht

Einen Fehler hast Du gefunden, aber wie viele `babbling idiot` hast Du noch in deinem System?
Such danach mal im Homematic-Forum.  
Häufig ist die Lösungsstartegie alle batteriebetriebenen Geräte stromlos zu machen oder die Batterien zu tauschen.

**Reply by The-Real-Lemonbiter** [reply]
Ne, das ist kein Babbling Idiot. WIe geschrieben: Sobald ich die HomematicIP Installation in HA deaktiviere geht der DC runter

**Reply by SukramJ** [reply]
Egal. Ich halt mich da raus. >15000 andere Nutzer haben dein Problem nicht.

**Reply by The-Real-Lemonbiter** [reply]
Verstehe... Tut mir auch leid... ich wünschte auch bei mir ging es glatt.

**Reply by The-Real-Lemonbiter** [reply]
Leider hat das Löschen so nicht vollständig funktioniert. Ich habe die Integration gelöscht, damit war auch unter HACS das Repository gelöscht. Dann habe ich überprüft, ob der angegeben Ordner in /config/custom_components gelöscht ist (ja, war nicht vorhanden) dann habe ich HA 2 mal neugestartet. Über HACS das Repo wieder hinzugefügt, alles angepasst und dann beim starten kam der Hinweis, das Gerät sei schon vorhanden. War also wieder da mit dem seinerzeit vergeben Namen und nicht dem neuen diesmal ausgewählten...

Zusatzfrage: Was genau bewirkt der Haken: "Heizgruppen aktivieren"? Denn den habe ich jetzt mal "ausgeschaltet" und HA neugestartet, einige aber NICHT alle Heizgruppen sind weg, 2 von 9 sind beharrlich da und bedienbar geblieben...
Da mein Problem ja scheinbar mit Heizgruppen (HM Wandthermometer, der ist ja auch der Gruppenkoordinator bei Heizgruppen) zu tun hat frage ich mich: Was genau macht die Funktion und wenn ich sie Ausschalte, warum sind dann einige Heizgruppen noch da und andere  nicht...

Danke und ich weiss, Du machst das alles hier ohne Kostendeckung. Ich muss das nicht kostenlos mit Dir machen. Ich zahle Dich gerne - es ist mein Hobby, ich würde das gerne zum Laufen bringen und wenn ich Dich zahlen darf, dann gerne!!!

**Reply by SukramJ** [reply]
Danke, ich nehme kein Geld, und schon gar nicht wenn ich das Problem nicht beeinflussen kann.

> "Heizgruppen aktivieren"?

Dann werden die Heizgruppen der CCU geladen. Wenn du das wieder deaktivierst musst Du nach einem Neustart ggf. die Entitäten manuell löschen.

Setz dir doch einfach parallel eine neu VM mit HA auf, und fang mal frisch an, dann sollten auch erst mal deine anderen Fehlermeldung aus dem Log weg sein.

**Reply by The-Real-Lemonbiter** [reply]
Lieber Sukram,
ich hoffe Du bist noch nicht fertig mit mir... :-)

Deiner Empfehlung folgend habe ich zunächst in der HM Installationso die komplette Firewall ausgeschaltet und die CCU neu gestartet, alle Ports sind offen. Dann habe ich neue VM aufgesetzt. Darin ist ausser den durch HA entdeckten Integrationen nichts manuell angelegt. Dann erst SSH und HACS installiert und final Deine Integration in der aktuellen stable

Leider am Ende das gleiche Chaos ... Und wieder einen Absturz, erst der Integration und dann von HA
Wenn Du mir weiter helfen magst, was ich super toll fände..  ich habe alles genau dokumentiert mit Screenshots etc. 
Hier die nicht ganz so Kurzfassung:

Es gibt ein paar recht interessante Fehlermeldungen... Final ist es so: die HMIP Integration  meldet timeout zu HMIP aber die CCU ist über die Website erreichbar und bedienbar,  ioB macht keine Fehlermeldungen, ich kann auch da alles bedienen. 

Wenn ich zu dem Zeitpunkt über die Integration in HA bspw. einen Rolladen steuere, dann fährt der auch, aber die Entität bekommt kein Feedback zur Position... das führt ein paar Minuten später dazu, dass sich die Integration aufhängt, den DC der CCU hochtreibt und die Integration lässt sich nicht beenden. Es kommt zu einer Fehlermeldung, dass die Entität sich nicht beenden lässt, dann hängt sich ganz HA auf und braucht ca. 5 Minuten bis es wieder erreichbar ist mit Fehlermeldung dass u.a. die Datenbank nicht OK ist. 
Und da sich die Integrität nicht beenden ließ vor dem Absturz von HA, wird sie nach dem Neustart automatisch neugestartet und kann sich dann nicht aktualisieren...

Soll ich das dokumentieren und Dir als PDF senden?
Ich finde HA generell so klasse, habe schon viele Integrationen am laufen, habe auch schon einige Dashboards gebaut und will nicht zurück zu ioB... BITTE HILF :-)
LG
LEM

**Reply by SukramJ** [reply]
Warum hängst du nicht einfach das Log als Datei an? Mit langen Texten ist dir nicht geholfen.
Auch wenn Du es nicht hören willst: Das Verhalten deiner CCU ist nicht normal. Du hast ein Problem mit deiner CCU, also zieh deine Schlüsse was man da wohl machen sollte.

**Reply by SukramJ** [reply]
> Und wieder einen Absturz, erst der Integration und dann von HA

Damit kann ich nichts anfangen, da ich sowas noch nie hatte. Hierzu könnten zusätzlich die HA Supervisor logs Aufschluss geben.

**Reply by The-Real-Lemonbiter** [reply]
Lieber SukramJ, ich glaube nun (mal wieder) es wirklich gelöst zu haben. Seit gestern Nacht läuft es ohne relevante Fehlermeldungen, der DC meiner RM steigt nicht wesentlich, auch nicht nach Neustarts der Inegration...  
Ich analysiere und wenn ich es wirklich final weiss, dann sage ich Dir sehr gerne, wo der Konfig Fehler auf meiner Seite lag. 
Vorab: Ich glaube, es liegt an der MTU Size meiner VM. Diese habe ich für die VM von HA nun reduziert und ...
Ich melde mich ASAP wenn ich sicher bin, dass es gelöst ist... und übrigens: DIe alte HA Installation ist die, die jetzt fehlerfrei läuft) HA :-)

**Comment by The-Real-Lemonbiter** [comment, marked as answer]
Sodele, lieber SukramJ, Deinem Namen folgend sage ich ein großes Dankeschön für Deine Entwicklung und... ich habe die Sache für mich endgültig gelöst!

Es handelte sich wohl um eine teils missverstandene Dokumentation (ja, das Problem sitzt wie immer vor dem Monitor... wie meine Entwickler immer sagen) und speziell an der Anzahl der Geräte / Entitäten / Systemvariablen. Um genau zu sein:

Was hat das Problem gelöst? Das Deaktivieren einiger Systemvariablen.

Zur Erklärung vorab: In meiner CCU gibt es insgesamt 143 Systemvariablen, von denen 25 regelmäßig (quasi alle paar Sekunden) aktualisiert werden. Das sind die Werte meines PV-Wechselrichters. Diese werden in ioBroker nach Änderung der Werte per Skript in die Systemvariablen der CCU geschrieben... Ja, WAHNSINN... ich weiss und baue das nun um!

Hier meine Analyse:

In den letzten Tagen habe ich ein neues, leeres Home Assistant OS System in Proxmox aufgesetzt. Nach der Installation von HACS habe ich einen Snapshot erstellt, um schneller testen zu können.

Bei der ersten Installation Deiner Integration hatte ich die Option „Systemvariablen-Scan aktivieren“ mit dem vorgegebenen Intervall von 30 Sekunden aktiviert. Nach dem Aktivieren des Dienstes stieg der Duty Cycle (DC) der CCU ins Unermessliche (DC = 99). Pong-Fehler, CCU nicht erreichbar…

Daraufhin habe ich die Integration deaktiviert, und der DC kehrte nach etwa 50 Minuten wieder auf Normalwerte zurück (DC = 11-15).

Dann habe ich den Snapshot (vor der Installation Deiner Integration) von Home Assistant wiederhergestellt und die Integration erneut installiert – dieses Mal ohne den Haken bei „Systemvariablen-Scan aktivieren“. Das Ergebnis: ein kleiner Anstieg des DC auf 26, der dann aber schnell wieder zurückging. Ich ließ das System ein paar Stunden so laufen – und keine Probleme!

Gegentest: Ich habe die oben genannten 25 Systemvariablen manuell wieder aktiviert. Ergebnis nach 10 Minuten: Pong-Fehler, DC steigt wieder auf 99… Da war mir klar: Es sind die Systemvariablen. Zum Test habe ich dann nur solche Systemvariablen aktiviert, die sich nur selten ändern (Türkontakte, Lichtschalter). Kein Fehler, nur ein minimaler und nicht relevanter DC-Anstieg beim Aktivieren der Integration.

Ich sehe ein, dass mein System mit 113 Geräten und 1300 Entitäten, wovon alleine 143 Systemvariablen sind, wohl eher nicht der Standard ist. Und vielleicht war es auch nicht die beste Idee, alle Systemvariablen zu aktivieren. Aber ich bin sehr froh, dass ich das Problem lösen konnte, denn der Umzug zu Home Assistant ist für mich ein großes Ziel, und die Integration von Homematic ist dabei ein wichtiger Baustein. Dank Deiner Integration und meiner wachsenden Lernkurve komme ich diesem Ziel nun einen Schritt näher!

Ich weiß, Du hast geschrieben, dass die Integration nichts macht außer die Werte abzufragen, aber die CCU scheint das in meinem Fall wohl anders zu sehen – zumindest nicht bei diesen schnell wechselnden SysVars. Ich habe es jetzt so gelöst, dass ich mir die Werte, die ich in Home Assistant benötige, per MQTT aus ioBroker übermittle und nur noch 30 Systemvariablen aktiviert habe. Der Duty Cycle pendelt sich seither wieder bei durchschnittlich 13-15 ein. Vielleicht lässt sich so etwas systemseitig in Deiner Integration abfangen? Falls mal wieder jemand mit so einem „System-Sonderling“ wie meinem auf Dich zukommt, hast Du vielleicht einen Hinweis für ihn.

Darf ich noch eine Kleinigkeit fragen? Drei der Systemvariablen, die ich in Home Assistant nutze, sind die Zustände von Türen. In der CCU sehe ich den Wert der Variablen als „true“ oder „false“. In Deiner Integration kommen sie jedoch als „Aus“ und „An“ an. Kann ich das irgendwie ändern?

Lieben Gruß und Danke an Dich!

**Reply by SukramJ** [reply]
Bitte lass das _lieber_ weg, und eine weitere persönliche Ansprache ist auch nicht notwendig.

Der DC steigt nur wenn die CCU mit den Geräten funkt.
Wenn Systemvariablen abgerufen werden, zumal alle in einem Rutsch abgerufen werden, findet keine Gerätekommunikation statt.
Aber wenn es für dich läuft, ist es auch egal.

> Drei der Systemvariablen, die ich in Home Assistant nutze, sind die Zustände von Türen. In der CCU sehe ich den Wert der Variablen als „true“ oder „false“. In Deiner Integration kommen sie jedoch als „Aus“ und „An“ an. Kann ich das irgendwie ändern?

Boolsche Sysvars werden als `binary_sensor` angelegt, die standard mäßig als an/aus angezeigt werden.
Ändere kann man das, indem man in den HA-Einstellungen des `binary_sensor` die Konfiguration ändert.
